File: Ridy-taxi-app/ios/common/Models/Objects/Driver.swift
//
//  Driver.swift
//
//  Copyright (c) Minimalistic Apps. All rights reserved.
//

import Foundation
import MessageKit

public final class Driver: Codable, SenderType {
    public var senderId: String { get {
            return "d\(String(self.id!))"
        }
    }
    
    public var displayName: String { get {
         return "\(self.firstName ?? "") \(self.lastName ?? "")"
        }
    }
    
    // MARK: Properties
    public var id: Int?
    public var firstName: String?
    public var lastName: String?
    public var certificateNumber: String?
    public var mobileNumber: Int64?
    public var email: String?
    public var balance: Wallet?
    public var car: Car?
    public var carColor: String?
    public var carProductionYear: Int?
    public var carPlate: String?
    public var carMedia: Media?
    public var status: Status?
    public var rating: Int?
    public var reviewCount: Int?
    public var media: Media?
    public var gender: String?
    public var accountNumber: String?
    public var bankName: String?
    public var bankRoutingNumber: String?
    public var bankSwift: String?
    public var address: String?
    public var infoChanged: Int?
    public var documentsNote: String?
    public var services: [Service]?
    public var documents: [Media]?
    
    public enum Status: String, Codable {
        case Offline = "offline"
        case Online = "online"
        case InService = "in service"
        case Blocked = "blocked"
        case Disabled = "disabled"
        case PendingApproval = "pending approval"
        case WaitingDocuments = "waiting documents"
        case SoftReject = "soft reject"
        case HardReject = "hard rejet"
    }
}

File: Ridy-taxi-app/android/common/src/main/java/com/innomalist/taxi/common/models/Driver.kt
package com.innomalist.taxi.common.models

import com.innomalist.taxi.common.Config
import com.squareup.moshi.Json
import com.squareup.moshi.JsonClass
import com.stfalcon.chatkit.commons.models.IUser
import java.sql.Timestamp

@JsonClass(generateAdapter = true)
data class Driver(
    var accountNumber: String?,
    var bankName: String?,
    var bankRoutingNumber: String?,
    var bankSwift: String?,
    var media: Media? = null,
    var carMedia: Media? = null,
    var carPlate: String? = null,
    var address: String? = null,
    var gender: Gender? = null,
    val rating: Int? = null,
    var firstName: String? = null,
    var lastName: String? = null,
    var carColor: String? = null,
    var certificateNumber: String? = null,
    var carProductionYear: String? = null,
    val id: Long? = null,
    val mobileNumber: Long? = null,
    val car: Car? = null,
    var email: String? = null,
    var status: Status? = null,
    val documentsNote: String? = null,
    val wallet: List<WalletItem>? = null,
    val documents: List<Media>? = null,
    var services: List<Service>? = null): IUser {


    override fun getId(): String {
        return id.toString()
    }

    override fun getName(): String {
        return lastName!!
    }

    override fun getAvatar(): String? {
        return if (media == null) null else "${Config.Backend}${media!!.address}"
    }

    enum class Status(val value: String) {
        @Json(name="offline")Offline("offline"), @Json(name="online")Online("online"), @Json(name="in service")InService("in service"), @Json(name="blocked")Blocked("blocked"), @Json(name="pending approval")PendingApproval("pending approval"), @Json(name="waiting documents")WaitingDocuments("waiting documents"), @Json(name="soft reject")SoftReject("soft reject"), @Json(name="hard reject")HardReject("hard reject");

        companion object {
            operator fun get(code: String): Status {
                for (s in values()) {
                    if (s.value == code) return s
                }
                return Offline
            }
        }
    }
}